### ISSUE TITLE
Using a Delayed Future in Widget test stops the test from completing

### ISSUE THREAD

**Issue by maks** [felangel/mocktail#19]
@felangel thanks for this fantastic package, you've saved mocking for us in the NS age!

I've have run into an issue though and I'm not sure if this is a case of my not doing something correctly but I've found that trying to use `Future.delayed()` in a mocked method in a widgetTest causes the test to hang and never complete.

My code is:
```dart

testWidgets('home page shows progress indicator when loading',
      (tester) async {
    final api = MockApiService();
    final lvm = LoginViewModel(api);

    when(api).calls(#requestLoginCode).thenAnswer((_) async {
      await Future<void>.delayed(Duration(seconds: 1));
      return APIResult.Error;
    });

    final w = ProviderScope(
      overrides: [loginProvider.overrideWithProvider(Provider((ref) => lvm))],
      child: HomePage(),
    );

    await tester.pumpWidget(wrapForTest(w));

    await lvm.requestCode('12345');

    await tester.pump();

    await tester.pump(Duration(seconds: 2));

    final hasSpinner = tester.any(find.byType(CircularProgressIndicator));
  
    expect(hasSpinner, true);
  });
```

where `requestCode()` calls the `ApiService` objects `requestLoginCode()` method which returns a Future that completes once its network API call (in the real, nnon mocked ApiService class) completes.

If I remove the `Future.delayed()` everything works as expected, BUT I want the delay in there because I'm trying to test that a `CircularProgressIndicator` is displayed during the period before the Future returned by `requestLoginCode` completes.

I've included the Riverpod setup code, but I don't think its relevant here.

My suspicion is this is maybe due to widgetTests running inside a FakeAsync zone though according to [this SO question](https://stackoverflow.com/questions/52472836/how-to-widget-test-futurebuilder-with-unresolved-future) this approach should work (or at least used to work with mockito).

**Comment by felangel** [comment]
Hi @maks 👋 
Thanks for opening an issue!

Are you still able to reproduce this in the latest release (v0.0.2-dev.2)? Thanks! 🙏

**Comment by felangel** [comment]
@maks any updates or can this be closed?

**Comment by maks** [comment]
@felangel sorry for the slow reply, its been a long weekend holiday here.
I've upgraded to `0.0.2-dev.5` and changed my test to now use new type-safe syntax:
```
    when(() => authRepo.requestOtp('12345')).thenAnswer((_) async {
      await Future<void>.delayed(Duration(seconds: 1));
      return AuthResult.error(400, 'test error');
    });
```

But unfortunately still seeing the test hang.

**Comment by felangel** [comment]
@maks no worries thanks for getting back to me!

Just to clarify, you said this was not a problem using `mockito`?

I think this is expected because in a widget test you will need to use [`tester.runAsync(() {...})`](https://api.flutter.dev/flutter/flutter_test/WidgetTester/runAsync.html) in this situation.

**Comment by maks** [comment]
@felangel actually I realised when I looked again at my test that I was making it more complicated than it needed to be as I could just override my providers state and not even need to worry about the async behaviour. 

I still think there maybe an issue with async handling as I would have expected that FakeAsync to work with the Future I was declaring inside the FaskAsync Zone, but I'll close this for now until I get a chance to see if I can write a simple standalone test to reproduce the issue. Thanks again for looking at it!
